### ISSUE TITLE
README please?

### ISSUE THREAD

**Issue by manighahrmani** [MysteryCrystal/TicTacToe#1]
Hi Allan. Can you please add a README to this project and describe how I can recreate your issue to suggest a fix?

**Comment by MysteryCrystal** [comment]
yeah, what do you mean? because this was just an issue visible by reading the debug prints.  I've since updated the backpropagation in the Networks.dart but still no luck at getting it to work.

**Comment by manighahrmani** [comment]
OK describe to me what u see in the debug prints (tell me step-by-step instructions on how I can reproduce it). 
If you are serious and want to learn AI, I would ask you first to complete [this tutorial](https://pytorch.org/tutorials/beginner/blitz/autograd_tutorial.html) for me (it takes at most a couple of hours).

**Comment by MysteryCrystal** [comment]
ok will do. I've been studying it at a lower level for a while, watching videos from 3b1b and others. I'll complete the tutorial as well.

**Comment by MysteryCrystal** [comment]
I've read through it but all of this is for python. I've also been trying to figure out why neural networks are built the way they are rather than simply accepting information as being true. Maybe I don't understand it fully but why can't you just take the derivative of the cost function with respect to its weights and set it to 0? wouldn't that give you the most accurate value for the updated weight? Or is there some issue with it overfitting a single set of data as a result?

**Comment by MysteryCrystal** [comment]
I take computer science. And of course dart really isn't the tool to be doing this but Nadim set me a challenge to create a tictactoe algorithm and I've been wanting to learn the fundamentals of neural networks in practice so I included it. This is also the reason I'm not doing it in python even though it does have numpy I'm enjoying building everything from the ground up. Looking at the list you sent I've actually already built a few discord bots in python. Is that something the university server actually needs? because a quiz app could be built in a few hours.

**Comment by MysteryCrystal** [comment]
well I've built a moderation bot in python which was paired with a html website and a SQL database as well as the paypal API in php. But the development of the python API has been turbulent. I had to swap to a fork because the creator got annoyed at the discord dev's direction. I had to then swap back when he came back and then discord stopped supporting prefix commands, opting for the more secure slash commands. so it does make it a bit difficult to get back into it but I can try.

**Comment by MysteryCrystal** [comment]
In fact I have one of my old discord bot templates here: https://github.com/MysteryCrystal/Bot-Template

**Comment by manighahrmani** [comment]
> well I've built a moderation bot in python which was paired with a html website and a SQL database as well as the paypal API in php. But the development of the python API has been turbulent. I had to swap to a fork because the creator got annoyed at the discord dev's direction. I had to then swap back when he came back and then discord stopped supporting prefix commands, opting for the more secure slash commands. so it does make it a bit difficult to get back into it but I can try.

I need to know about your career goals. If it wasn't related I won't bother you but it would be nice if you could help develop a quiz bot at first or maybe a role one. The moderation one can be more time-consuming because you'd have to interview Matt and Rich to see exactly what they want.

> In fact I have one of my old discord bots here: https://github.com/MysteryCrystal/Mayumi-Bot. It was meant to be collaborative and there is some data there that should be kept private such as the token or database sign ins but as everything is offline it shouldn't be an issue.

Am I safe to assume that this bot essentially helps with moderation by providing commands but not scanning/flagging messages? My current students managed to hook it up to Azure or GCP AI resources so we get these scores to flag automatically.

**Comment by manighahrmani** [comment]
Can you message me on Discord pls? It's easier than here.
